olxcleaner/loader/xml.py

```diff
@@ -109,9 +109,9 @@ def read_course(edxobj, node, directory, filename, errorstore, htmlfiles, pointe
                     empty = False
 
     # Check for a pointer tag
+    node_filename = os.path.splitext(os.path.split(filename)[-1])[0]
     if empty and edxobj.is_pointer(node.attrib):
-
-        if pointer:
+        if pointer and node.attrib.get('url_name') == node_filename:
```

Comment:
You need to catch the case where the url_name is not the node filename here and raise a different error.

Reply:
Note that this will probably require creating a new error message.

Reply:
@jolyonb
right, right, sound like you mean if the node is a pointer and 1. pointer URL is same as filename (which is initially this pr covered) and 2. pointer URL is different than the filename (like the case you mentioned).

I want to add -- if we do not catch the 2nd case, we would get `URLNameMismatch` Error.

Reply:
I suspect if the file actually exists, it'll be treated as a pointer tag and will go through without an error (currently).

Reply:
(pointer tags aren't checked for urlnamemismatch)